src/main/java/org/folio/helper/BindHelper.java

```diff
@@ -112,11 +138,16 @@ private Future<Map<String, List<Piece>>> createItemForPiece(Map<String, List<Pie
       .map(PoLineCommonUtil::convertToCompositePoLine)
       .compose(compPOL ->
         inventoryItemManager.createBindItem(compPOL, holdingIds.get(0), bindPiecesCollection.getBindItem(), requestContext))
-      .map(itemId -> {
-          piecesGroupedByPoLine.get(poLineId).forEach(piece -> piece.setItemId(itemId));
-          return piecesGroupedByPoLine;
+      .map(newItemId -> {
+        // Transfer requests if flag is on
+        if (TRANSFER.equals(bindPiecesCollection.getTransferRequests())) {
```

Comment:
getTransderRequests sounds like collection of objects.
We should rename this to somethig like this: requestsAction: TRANSFER, CANCEL, DO_NOTHING

Reply:
Renamed, good point